include-blang/cooperative_groups.h

```diff
@@ -0,0 +1,32 @@
+#ifndef _COOPERATIVE_GROUPS_H_
+#define _COOPERATIVE_GROUPS_H_
+
+namespace cooperative_groups {
+  class grid_group {
+    public:
+      __forceinline__ __device__ void sync() const {
+          bugle_grid_barrier();
+      }
+  };
+  
+  class thread_block {
+    public:
+      __forceinline__ __device__ void sync() const {
+          bugle_barrier(true, true);
+      }
+  };
+  
+  __forceinline__ __device__ grid_group this_grid();
```

Comment:
As this function as no body, `__forceinline__` can be omitted.
```suggestion
  __device__ grid_group this_grid();
```

Reply:
Sure, will make this change.